gnomad/utils/constraint.py

```diff
@@ -125,3 +127,274 @@ def collapse_strand(
+    ).aggregate(
+        low_coverage_obs_exp=hl.agg.sum(all_low_coverage_ht.observed_variants)
+        / (
+            high_coverage_scale_factor
+            * hl.agg.sum(
+                all_low_coverage_ht.possible_variants * all_low_coverage_ht.mu_snp
+            )
+        )
+    )
+    coverage_model = build_coverage_model(
+        low_coverage_ht,
+        low_coverage_obs_exp="low_coverage_obs_exp",
+        log_coverage="log_coverage",
+    )
+    # TODO: consider weighting here as well
+
+    return coverage_model, plateau_models
+
+
+def build_plateau_models_pop(
+    ht: hl.Table,
+    cpg: str,
+    mu_snp: str,
+    observed_variants: str,
+    possible_variants: str,
+    pop: str,
+    length: int,
+    weighted: bool = False,
+) -> hl.Struct:
+    """
+    Build plateau model for each `pop` population to calibrate mutation rate to compute predicted proportion observed value.
+
+    The x and y of the plateau models:
+    - x: `mu_snp` - mutation rate
+    - y: proportion observed (observed_{pop}' / 'possible_variants')
+
+    :param ht: High coverage Table.
+    :param cpg: The annotation name of booleans that determine whether the site is CPG site or not.
+    :param mu_snp: The annotation name of the mutation rate.
+    :param observed_variants: The annotation name of the observed variant counts for each combination of keys in `ht`.
+    :param possible_variants: The annotation name of the possible variant counts for each combination of keys in `ht`.
+    :param pop: List of populations. Defaults to None.
+    :param length: The minimum array length of variant counts array in downsampings for `pop`.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants', defaults to False.
+    :return: A Dictionary of intercepts and slopes for plateau models of each population.
+    """
+    agg_expr = {
+        pop: [
+            hl.agg.group_by(
+                ht[cpg],
+                hl.agg.linreg(
+                    ht[observed_variants][i] / ht[possible_variants],
+                    [1, ht[mu_snp]],
+                    weight=ht[possible_variants] if weighted else None,
+                ).beta,
+            )
+            for i in range(length)
+        ]
+    }
+    return ht.aggregate(hl.struct(**agg_expr))
+
+
+def build_plateau_models_total(
+    ht: hl.Table,
+    cpg: str,
+    mu_snp: str,
+    observed_variants: str,
+    possible_variants: str,
+    weighted: bool = False,
+) -> hl.Struct:
+    """
+    Build plateau model for all release samples without downsamplings to calibrate mutation rate to compute predicted proportion observed value.
+
+    The x and y of the plateau models:
+    - x: `mu_snp` - mutation rate
+    - y: proportion observed ('observed_variants'/ 'possible_variants')
+
+    :param ht: High coverage Table.
+    :param cpg: The annotation name of booleans that determine whether the site is CPG site or not.
```

Comment:
```suggestion
    :param cpg: The name of the Boolean noting whether a site is a CPG site.
```

Reply:
also, if this is the annotation name, should this be `cpg_str` or something similar?